### PR TITLE
Scrubber Clog Revisit: It Uses Vents Now And Is Way Dirtier And Nastier

### PR DESCRIPTION
**Pull request by Rhials**
## About The Pull Request

scrubber to vent

A long long time ago, when I was but a wee coder, I made the Scrubber Clog event. I wasn't great at coding then, but about a year later I have returned to finish what I started.

This PR contains a full code/balance/feature update for the event, bringing it up to what fledgeling coder Rhials had envisioned but wasn't experienced enough to do. This contains no fundamental or conceptual changes, just some TLC.

Since everything's in a new file and there isn't a diff to compare things with, I'll try to give a quick summary of everything:

FEATURES:

- _The clog event now uses the ventilation pumps instead of scrubbers. It is now the ventilation clog event. Rejoice._
- Clogs now make a filthy mess when they occur, and make noise when spawning mobs.
- As more mobs spawn, even more of a mess is made. Different clog event intensities will produce different messes.
- The mob roster has been updated to include the Mobroach, Mothroach, and Viscerator for the Strange clog event, and the Hauberroach for the Major clog event.

CODE:

- The event is now fully self-contained, without requiring any code on the vent itself to work.
- plunger_act() now has its own signal.

BALANCE (This is just a bunch of tiny insignificant number tweaking don't worry): 

- More mobs can be spawned at once by the event. This shouldn't have any real impact on how dangerous the minor/major clog events are, but might make the critical/strange events a smidge more dangerous.
- Max occurrences have been increased. The minor/major clog events are no longer limited to only three per round.
- There is now a slight chance that a clog event will not announce itself.
## Why It's Good For The Game

The clog event producing a big mess was something I had wanted to do in the original PR, but I either forgot or was incapable at the time. Now, there's even more of that "janitor content" I had originally promised. Get cleaning!

The code improvements speak for themselves -- Having the produce_mob() and clog()/unclog()/clogged on the scrubber was something that really irked me once I started learning more about how events work. 

Since the minor/major events are mostly fluff that pose no major impact on the round, I figured having them occur more than just a few times per round would be fine. More mobs at once means more filth, and more of a problem if you leave the clog to fester for too long.

(Oh, and on the whole "scrubbers to vents" change -- The clog event was added with inspiration from the then-removed scrubber _overflow_ event. The overflow event was readded a few months later, and since then I have been overwhelmed with the urge to make this change because it felt weird having two "stuff comes out of the scrubbers for no real reason" events.)
## Changelog
:cl: Rhials
add: The scrubber clog event is now the ventilation clog event. It uses vents instead of scrubbers now. It is now the ventilation clog event.
add: Ventilation clogs now produce a lot of filth in addition to produced mobs, and make the surrounding area dirty.
balance: Ventilation clogs may produce more mobs at once, and have a higher limit on how many can occur per round.
code: The ventilation clog event is now self-contained, and handles all logic/behavior in the event itself (rather than on the targeted vent).
/:cl:

### CODE REVIEW

**Comment by MTandi** [comment]
It was very weird when I cancelled the random event but the spiders kept coming from the scrubbers... Ugh

File: code/game/objects/objs.dm [hunk 1]
```diff
@@ -334,7 +334,7 @@ GLOBAL_LIST_EMPTY(objects_by_id_tag)
 	return ..()
 
 /obj/proc/plunger_act(obj/item/plunger/P, mob/living/user, reinforced)
-	return
+	return SEND_SIGNAL(src, COMSIG_PLUNGER_ACT, P, user, reinforced)
```
Comment (san7890) [inline code/game/objects/objs.dm:337]: would you mind getting rid of the single-letter variable while you are here

File: code/modules/events/vent_clog.dm [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+/datum/round_event/vent_clog/proc/get_vent()
+	var/list/vent_list = list()
+	for(var/obj/machinery/atmospherics/components/unary/vent_pump/vent in GLOB.machines)
+		var/turf/vent_turf = get_turf(vent)
+		if(vent_turf && is_station_level(vent_turf.z) && !vent.welded)
+			vent_list += vent
+
+	if(!length(vent_list))
+		kill()
+		CRASH("Unable to find suitable vent.")
```
Comment (san7890) [inline code/modules/events/vent_clog.dm:99]: shouldn't this go before the forloop, in order to be a neat early return

File: code/modules/events/vent_clog.dm [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+	for(var/turf/nearby_turf in view(2, get_turf(vent)))
+		if(isopenturf(nearby_turf) && prob(85))
+			new /obj/effect/decal/cleanable/dirt(nearby_turf)
+
+	produce_mob()
+
+///Clears the signals related to the event, before we wrap things up.
+/datum/round_event/vent_clog/proc/clear_signals()
+	UnregisterSignal(vent, COMSIG_PARENT_QDELETING)
+	UnregisterSignal(vent, COMSIG_PLUNGER_ACT)
```
Comment (san7890) [inline code/modules/events/vent_clog.dm:192]:
```suggestion
	UnregisterSignal(vent, list(COMSIG_PARENT_QDELETING, COMSIG_PLUNGER_ACT))
```

**Review by san7890 — CHANGES_REQUESTED** [review]

File: code/modules/events/vent_clog.dm (hunk 2 above)
Comment (Rhials) [reply code/modules/events/vent_clog.dm:99]: the forloop populates the list with all valid scrubbers. It is here to make sure the list has scrubbers in it, so we don't pick/return from an empty list.

File: code/modules/events/vent_clog.dm [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+	min_players = 15
+	max_occurrences = 3
+	earliest_start = 25 MINUTES
+	description = "Really dangerous mobs climb out of a vent."
+	min_wizard_trigger_potency = 3
+	max_wizard_trigger_potency = 6
+
+/datum/round_event/vent_clog/critical/setup()
+	. = ..()
+	spawn_delay = rand(15,25)
```
Comment (carlarctg) [inline code/modules/events/vent_clog.dm:239]:
```suggestion
	spawn_delay = rand(1.5, 2.5) SECONDS
```

Comment (Rhials) [reply code/modules/events/vent_clog.dm:239]: This value is a measure of event ticks, not deciseconds.

**Comment by khevy** [comment]
I didn't see this pull until now.  Awesome seeing these in game for real

**Pull request merged**